kratos/python_scripts/python_solver.py

```diff
@@ -157,6 +157,8 @@ def _ImportModelPart(self, model_part, model_part_import_settings):
             from restart_utility import RestartUtility
             RestartUtility(model_part, self._GetRestartSettings(model_part_import_settings)).LoadRestart()
             self.print_on_rank_zero("::[PythonSolver]::", "Finished loading model part from restart file.")
+        elif(input_type == "use_input_model_part"):
```

Comment:
why is this needed?
In such cases I would not call `ImportModelPart`

I suggest doing sth like we do in [StructuralMechanics](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L204-L208) to not do modifications in this class

Reply:
In both the Buoyancy solver and the CHT solver, the thermal fluid model part is set from the CFD solver one. Then, the import is skipped in this cases. However, if you are solving a simple heat transfer problem, you need to import the model part as usual. To avoid reimplementing the entire ImportModelPart() method in the convection-diffusion solver including this if statement, we decided to add this option. IMO it does not harm and might be useful for similar developments.

Reply:
@philbucher if we decide to go with what you are doing in the StructuralMechanics, I'd suggest moving that (the member variable and the auxiliary function) to the base class.

Reply:
Yep whatever will be decided here I will also adapt in StructuralMechanics (FYI @dbaumgaertner @armingeiser )

Reply:
@philbucher i am not clear of what is your proposal. could u explain it a bit here?

Reply:
FYI @MFusseder

Reply:
@RiccardoRossi I think I like your solution more, so discard my comment
having a special `input_type` that does nothing (i.e. not import the ModelPart from a file) seems like a good solution on second thought
personally I think the name `use_existing_model_part` is more clear, but `use_input_model_part` would also be ok

once this is merged I will adapt StructuralMechanics to what we agree on here
@armingeiser @dbaumgaertner @MFusseder this means that you will have to do a minor modification in the ShapeOptApp. I will let you know

@RiccardoRossi To sum up, I am fine with the changes in this PR, from my side no separate PR is necessary